include/alpaka/core/Cuda.hpp

```diff
@@ -177,15 +177,44 @@ namespace alpaka
                 int const & line)
             -> void
             {
-                if(error != CUDA_SUCCESS)
+                if(error == CUDA_SUCCESS)
+                    return;
+
+                char const * cu_err_name = nullptr;
+                char const * cu_err_string = nullptr;
+                CUresult cu_result_name = cuGetErrorName(error, &cu_err_name);
+                CUresult cu_result_string = cuGetErrorString(error, &cu_err_string);
+                std::string sError;
+                if( cu_result_name == CUDA_SUCCESS && cu_result_string == CUDA_SUCCESS )
                 {
-                    std::string const sError(std::string(file) + "(" + std::to_string(line) + ") " + std::string(desc) + " : '" + cudaGetErrorName(static_cast<cudaError_t>(error)) +  "': '" + std::string(cudaGetErrorString(static_cast<cudaError_t>(error))) + "'!");
+                    sError = std::string(file)
+                        + "(" + std::to_string(line) + ") "
+                        + std::string(desc) + " : '"
+                        + std::string(cu_err_name) +  "': '"
+                        + std::string(cu_err_string) + "'!";
+                } else {
+                    // cuGetError*() failed as well, so print different error message
+                    sError =
+                        std::string(__FILE__)
+                        + "(" + std::to_string(__LINE__) + ") "
```

Comment:
I do not get what those `__FILE__` and `__LINE__` macros will get us. They do not point to any useful line. I would only use the file and line from the original error.
You still only print the available information when both methods were successful. I would prefer to have both appended individually.

Reply:
ok, no problem, I am going to change that.